### PR TITLE
fix: 10315 add logs to grant approval calls (ERCPrecompileSuite fuzzy match)

### LINKED ISSUE #10315: Enable fuzzy record matching for ERCPrecompileSuite
Fixed by:

- https://github.com/hashgraph/hedera-services/pull/11164
- https://github.com/hashgraph/hedera-services/pull/11178
- https://github.com/hashgraph/hedera-services/pull/11179
- https://github.com/hashgraph/hedera-services/pull/11183
- https://github.com/hashgraph/hedera-services/pull/11185

### PR DESCRIPTION
**Pull request by JivkoKelchev**
**Description**:

This PR modifies some NFT checks to grant approval calls to match mono service. 
Also, it adds logs for these calls.

**Related issue(s)**:

#10315 

Fixes specs:
someErc721ApproveAndRemoveScenariosPass
someErc20ApproveAllowanceScenarioInOneCall
someErc20ApproveAllowanceScenariosPass
erc721TokenApprove
erc20Approve

**Notes for reviewer**:

Links to mono service implementation:
[NFT checks](https://github.com/hashgraph/hedera-services/blob/4cb49be42ae037affbf14103a2fafb8c6d2ed7ae/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/ApprovePrecompile.java#L159-L166)
[Logs](https://github.com/hashgraph/hedera-services/blob/4cb49be42ae037affbf14103a2fafb8c6d2ed7ae/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/ApprovePrecompile.java#L283-L301)

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/grantapproval/AbstractGrantApprovalCall.java [hunk 1]
```diff
@@ -70,16 +67,8 @@ public TransactionBody callGrantApproval() {
...
-                List<AccountApprovalForAllAllowance> accountApprovalForAllAllowances = enhancement
-                        .nativeOperations()
-                        .getAccount(ownerId.accountNum())
-                        .approveForAllNftAllowances();
-                if (accountApprovalForAllAllowances != null) {
-                    for (var approvedForAll : accountApprovalForAllAllowances) {
-                        if (approvedForAll.tokenId().equals(token)) {
-                            return buildCryptoApproveAllowance(approveDelegate(ownerId, approvedForAll.spenderId()));
-                        }
-                    }
```
Comment (mhess-swl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/grantapproval/AbstractGrantApprovalCall.java:82]: Why were these lines removed?

Comment (tinker-michaelj) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/grantapproval/AbstractGrantApprovalCall.java:82]: This checks will be done in `dispatch()` to `CryptoApproveAllowanceHandler`.

**Review by tinker-michaelj — APPROVED** [review]
The changes to explicitly externalize a result from the approval calls were actually unneeded; the reason certain records didn't appear in the snapshot was that their response codes are listed [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/src/main/resource/spec-default.properties#L113), which lists response codes mono would generally filter in ingest.

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @JivkoKelchev !

**Pull request merged**